City-of-Helsinki/drupal-helfi-rekry
# UHF-8535: Forced translations

# [UHF-8535](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8535)
Get rid of forced translations.

## What was done
This branch:
* Removed all backend logic for creating forced translations of job listings
* Cleaned up leftover code from react-app-to-hdbt -migration

HDBT branch
* Changed query logic to exclude forced translations
* All node languages are now included in the search by default
  * If language filter in active, only listings of that language are shown
  * If no language filter is active and node has translations, prefer the one in current language 

## How to install

Make sure your instance is up and running on correct branch.
* `git fetch; git checkout origin/UHF-8535-forced-translations`

Get the HDBT branch:
* `composer require drupal/hdbt:dev-UHF-8535-forced-translations`

Set up search:
* Run `make up`
* Run `make ps` and note which port your Elastic is running in (900X)
* Add this to your .env file: `ELASTIC_PROXY_URL=https://localhost:` + elastic port
* Run `make up` to finalize .env changes
* Run `drush sapi-rt; drush-sapi-c; drush sapi-i;`

Finally, run `make drush-cr`.

## How to test

This branch:
* Read through the code - we are only deleting stuff in this branch so not much testing is required. Is there something missed or seeming out of place?

HDBT:
* If you have a fresh dump from test, you can navigate to https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/etsi-avoimia-tyopaikkoja (otherwise add job search paragraph to a landing page)
* Try making some searches. Test for these things:
  * Searching without language filter active you should see results in all available languages
  * With language filter active, you should see results only in that language
* Test the language prioritization:
  * Find a job listing that has a swedish translation. If it does not already have a finnish translation, add one. In any case edit the finnish translation to have a distinct title so you can spot if easier.
  * Run `drush sapi-i` after editing to apply changes to index.
  * Search for the title of the job you just edited. Make sure you haven't filtered by language. Test that:
    * When viewing the search page in finnish, you see the finnish translation of the job listing
    * When viewing the search page in swedish, you see the swedish translation of the job listing

## Designers review



## Other PRs


* HDBT PR: https://github.com/City-of-Helsinki/drupal-hdbt/pull/700


[UHF-8535]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8535?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
